---
title: "Individual Report"
output: github_document
---
```{r,echo=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(pheatmap)
library(survival)
library(knitr)
```

# Group Project - individual report: 

Survival and Gene Expressions Analysis of Pancreatic adenocarcinoma (PAAD)

## A concise summary of contributions of each group member (1 pts)

Group tasks were divided into five steps, including project topic background information research, Data Dexcrption and Wrangling, Exploratory Data Analysis (EDA), Modeliing Building, and results interpretation. And our group members include Hassan Ali, Simran Samra, Sophia Li and me. Hassian and Simran are from Biology background, and one works in BC cancer and another has experience in Heart Lung Innovation. Sophia and I are from Statistical background. Both of us have strong data analysis coding stills but lack of the experience in analyze high dimensional Genomic data. 

Based on the each member's academic backgroup, Hassan and Simran are assigned to the background information research ("blue task"), on the other hand, Sophia and I mainly focus on the model building ("yellow task"). Data Dexcrption and Wrangling, EDA and results interpretation ("green tasks") were done all together. 

![group tasks](https://github.com/STAT540-UBC/zz_Xia-Lily_STAT540_2020/blob/master/project/plot/group%20tasks.png)

"blue task": 

1. research the backgroud of Pancreatic adenocarcinoma (PAAD) and introduce the topic to other group members
2. evaluate whether group objectives are doable or not

"green task":

1. Data Description and Wrangling: decribe the data variables (gene data and metadata) and sanitize the data
2. EDA: explore the data without any assumption. For instance, using density plots, boxplots and principal components analysis (PCA) to visualize the data and compare different cohort of subjects. 
3. Results interpratation: explain the EDA and statistical models outcome and summary analysis limitation

"yellow task":

1. survival analysis
2. vital status classification
3. Hypothesis test
4. Clustering samples and genes

## Your specific contributions and comments (1 pts)
As mention above, I mainly focus on data cleaning and data analyzing. Besides, I also help to maintain the Github repo. Overall, all tasks worked well but final results interpration and limitation discussion were not as what I expected. From my perspective, having members from genome or biology backgroud could help us not only interprate the results statistically, but also assist us to think the results in different aspects. For example, I know that although our models shows there is no statistical significant between patients' gene expression and vital status, it does not mean there is no association between them. Therefore, In addition to highly expressed genes, I hope I could have more information on what mutational genes we should mainly focus on based on backgroud research. This project is really good hand-on experience to apply the approaches that we have learnt in class and seminars, and it's fun to work with different background students. 

## Answer to one question specific to your project (3 points)

**Q: What is your rationale for mostly analyzing the top 10 genes, and not for instance genes that pass FDR, or some other number of top genes? How do you think this impacted your results?**

```{r, echo = F,warning=FALSE, message=FALSE}
metadata <- read.csv("~/Desktop/git_docs/zz_Xia-Lily_STAT540_2020/project/data/tcga_paad_clinical.csv", header = T)
toDelete <- seq(0, nrow(metadata), 2)
metadata <-  metadata[-toDelete, ]
load("~/Desktop/git_docs/zz_Xia-Lily_STAT540_2020/project/data/tcga_paad.RData")
genedata <- tcga
rownames(genedata) <- rownames(tcga)
colnames(genedata) <- str_sub(colnames(genedata),1,12)

metadata <- metadata %>% select(c(submitter_id, age_at_index, 
                         year_of_death,year_of_diagnosis, 
                         days_to_death, vital_status, race, gender, ajcc_pathologic_m, 
                         ajcc_pathologic_n, 
                         ajcc_pathologic_t, ajcc_pathologic_stage,
                         treatment_or_therapy, treatment_type))


DesMat <- model.matrix(~ vital_status, metadata)
expressionMatrix <- t(scale(t(genedata)))
dsFit <- lmFit(expressionMatrix, DesMat)
ebfit <- eBayes(dsFit)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# convert a factor into a numeric variable without loss information
metadata$year_of_death <- as.numeric(as.character(metadata$year_of_death))
metadata$year_of_death <- ifelse(is.na(metadata$year_of_death), 2014, metadata$year_of_death)

metadata$year_of_diagnosis <- as.numeric(as.character(metadata$year_of_diagnosis))
year <- metadata$year_of_death - metadata$year_of_diagnosis

metadata$days_to_death <- as.numeric(as.character(metadata$days_to_death))

# complete the survival time variable (day unit)
day <- rep(NA,length(metadata$days_to_death))
for(i in 1:length(metadata$days_to_death)){
  day[i] <- ifelse(is.na(metadata$days_to_death[i]), year[i]*365, metadata$days_to_death[i])
}
metadata$time <- day

metadata <- metadata %>% select(-c(year_of_death, year_of_diagnosis, days_to_death))
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# top 5000 genes
top5000 <- rownames(topTable(ebfit, number=5000))
topF <-  genedata %>% filter(rownames(genedata) %in% top5000) 
rownames(topF) <- top5000
topF1 <- t(topF)
topF2 <- rownames_to_column(as.data.frame(topF1), var = "submitter_id")

meta_gene5000 <- right_join(metadata, topF2, by = "submitter_id")


# top 10 genes 
top10 <- rownames(topTable(ebfit))
topTen <-  genedata %>% filter(rownames(genedata) %in% top10) 
rownames(topTen) <- top10
topTen1 <- t(topTen)
topTen2 <- rownames_to_column(as.data.frame(topTen1), var = "submitter_id")

meta_gene10 <- right_join(metadata, topTen2, by = "submitter_id")

# suggested mutational signature data (Kras, tp53, cdkn2a and smad4)
mutgene <- c("KRAS", "TP53", "CDKN2A", "SMAD4")
mutFour <-  genedata %>% filter(rownames(genedata) %in% mutgene) 
rownames(mutFour) <- mutgene
mutFour1 <- t(mutFour)
mutFour2 <- rownames_to_column(as.data.frame(mutFour1), var = "submitter_id")

meta_mut4 <- right_join(metadata, mutFour2, by = "submitter_id")

```

```{r, warning=FALSE, message=FALSE}
(tptb <- toptable(ebfit)) %>% kable()
```

We have analysis a cohort of pancreatic adenocarcinoma (PAAD) patients who were diagnosed between 2001 to 2014. There are 177 patients with average follow-up 655 days. We consider the latest death as the endpoint in the analysis. The patients were between the age of 35-88 years old. To identify genes that are more heavily expressed in PAAD patients that have longer survival time compared to those that have a shorter survival time, we used `limma` and `eBayes` to model the relationship between gene expression and vital status.

`lmFit()` function that we used carries out multiple linear regression on each gene, where the explanatory variable was *vital_status*. The multiple linear regressions were conducted to evaluate the association between gene expression values and patients' survival situations. Meanwhile, since we perform the statistical test on each gene, the the dangerouse of multiple testing should be considered. We controled the false discovery rate (FDR) using Benjamini Hochberg method, and `adj.P.Val` is the BH FDR values given by default. We expected limma could narrow down the range of genes.

However, all `adj.P.Val` are the same and close to 1. We're not getting any genes with adjusted p-values below 0.05, meaning that there is no genes pass the FDR, and this data set doesn't provide any differentially expressed genes at a FDR threshold of 0.05. 

```{r, echo=FALSE}
preDat <- as.data.frame(metadata[, c("vital_status")])
rownames(preDat) <- metadata$submitter_id 


data_to_plot <- as.dist(1-cor(topTen))
data_to_plot <- as.matrix(data_to_plot)
pheatmap(data_to_plot, cluster_rows = T, scale =  "none" , clustering_method = "ward.D2",
         clustering_distance_cols = "euclidean", show_rownames = F, show_colnames = F,
         main = "Top 10 gene correlation between samples",
         annotation = preDat)
```

```{r, echo=FALSE} 
data_to_plot <- as.dist(1-cor(topF))
data_to_plot <- as.matrix(data_to_plot)
pheatmap(data_to_plot, cluster_rows = T, scale =  "none" , clustering_method = "ward.D2",
         clustering_distance_cols = "euclidean", show_rownames = F, show_colnames = F,
         main = "Top 5000 gene correlation between samples",
         annotation = preDat)
```

```{r, echo=FALSE}
data_to_plot <- as.dist(1-cor(mutFour))
data_to_plot <- as.matrix(data_to_plot)
pheatmap(data_to_plot, cluster_rows = T, scale =  "none" , clustering_method = "ward.D2",
         clustering_distance_cols = "euclidean", show_rownames = F, show_colnames = F,
         main = "Four mutational genes correlation between samples",
         annotation = preDat)
```

These are the heatmaps for different selected genes correlation between samples. Since we cannot access the control cohort, the mutational signature genes cannot be detected. The suggested mutational genes are `KRAS`, `TP53`, `CDKN2A` and `SMAD4`. We can see that the shown sample clusters in all heatmaps might not due to different vital status.

No matter what number of top genes used to analyze, `adj.P.Val` are still constant and close to 1. However, an adjusted p-value of 1 doesn't mean there is no differential expression, it just means that this particular data set does not show any evidence of differential expression. The insufficient quality of the data set might because of hidden batch effects obscuring the true effect or other errors that we need to figure out. 

The rationale for mostly analyzing the top 10 genes is not because the target genes are 100 percent included in such a small range. But we thought our first step of finding the evidence of differential expression among 44,084 genes should start from the highly expressed genes. Furthermore, we are not able to handle the full set of genes in survival analysis, so top 10 genes maybe a good start. We use `step()` function based on Akaike Information Criterion (AIC) to further select genes in the top10, and fit the Cox porportional hazard model.

```{r, echo=FALSE}
dat2 <- meta_gene10 %>% drop_na()
dat2$vital_status <- as.numeric(dat2$vital_status)-1
```

```{r}
fit <- coxph(formula = Surv(time, vital_status) ~ THBS1 + NNMT + CREM + 
    OLFML2B, data = dat2)
summary(fit)
```

We found that increaseing the expression value of `THBS1`, `NNMT` will decrease the log hazard ratio (longer survival time) by 10.53 and 1.875 respectively, but if increase the expression value of `CREM` and `OLFML2B`, then log hazard ratio will  increase 1.647 and 2.500 respectively. The Cox model diagnostics was performed as follow: We use Martingale residual plot to check the functional form of the covariates in the Cox model. A approximate straight line in each plot indicates that a linear form is needed. We also check the constancy of the coefficients using Schoenfeld residual against a nonlinear scale of time with spline smoother.

```{r, echo=FALSE}
par(mfrow = c(2,2))
scatter.smooth(dat2$THBS1, residuals(fit), main ="residual plot",
               ylab = "martingale residuals", xlab = "THBS1 expression")
scatter.smooth(dat2$NNMT, residuals(fit), main ="residual plot",
               ylab = "martingale residuals", xlab = "NNMT expression")
scatter.smooth(dat2$CREM, residuals(fit), main ="residual plot",
               ylab = "martingale residuals", xlab = "CREM expression")
scatter.smooth(dat2$OLFML2B, residuals(fit), main ="residual plot",
               ylab = "martingale residuals", xlab = "OLFML2B expression")

par(mfrow = c(2,2))
plot(cox.zph(fit))
```



```{r eval=FALSE, include=FALSE, echo=FALSE}
# Top 10 gene
dat2 <- meta_gene10 %>% drop_na()
dat2$vital_status <- as.numeric(dat2$vital_status)-1

fit2 <- coxph(Surv(time, vital_status) ~ THBS1 + `CTD-2033D15.2` + NNMT + CREM + OLFML2B + CYP1B1 + KCNE4 + ITPRIP + `RP11-21L23.2` + CTGF ,
         data = dat2)

step(fit2)

fit <- coxph(formula = Surv(time, vital_status) ~ THBS1 + NNMT + CREM + 
    OLFML2B, data = dat2)
summary(fit)

par(mfrow = c(2,2))
scatter.smooth(dat2$THBS1, residuals(fit), main ="residual plot",
               ylab = "martingale residuals", xlab = "THBS1 expression")
scatter.smooth(dat2$NNMT, residuals(fit), main ="residual plot",
               ylab = "martingale residuals", xlab = "NNMT expression")
scatter.smooth(dat2$CREM, residuals(fit), main ="residual plot",
               ylab = "martingale residuals", xlab = "CREM expression")
scatter.smooth(dat2$OLFML2B, residuals(fit), main ="residual plot",
               ylab = "martingale residuals", xlab = "OLFML2B expression")

par(mfrow = c(2,2))
plot(cox.zph(fit))

# Top 5000 gene
dat3 <- meta_gene5000 %>% drop_na()
dat3$vital_status <- as.numeric(dat3$vital_status)-1

fit3 <- coxph(Surv(time, vital_status) ~ THBS1 + `CTD-2033D15.2` + NNMT + CREM + OLFML2B + CYP1B1 + KCNE4 + ITPRIP + `RP11-21L23.2` + CHSY1 + LUM + MEST + CYTL1 + MGP + PCDHB7 + OLFML1, data = dat3)

step(fit3)

# Four mutational genes
dat1 <- meta_mut4 %>% drop_na()
dat1$vital_status <- as.numeric(dat1$vital_status)-1

fit1 <- coxph(Surv(time, vital_status) ~ KRAS + TP53 + CDKN2A + SMAD4, data = dat1)
summary(fit1)

par(mfrow = c(2,2))
scatter.smooth(dat1$KRAS, residuals(fit1), main ="residual plot",
               ylab = "martingale residuals", xlab = "KRAS expression")
scatter.smooth(dat1$TP53, residuals(fit1), main ="residual plot",
               ylab = "martingale residuals", xlab = "TP53 expression")
scatter.smooth(dat1$CDKN2A, residuals(fit1), main ="residual plot",
               ylab = "martingale residuals", xlab = "CDKN2A expression")
scatter.smooth(dat1$SMAD4, residuals(fit1), main ="residual plot",
               ylab = "martingale residuals", xlab = "SMAD4 expression")

par(mfrow = c(2,2))
plot(cox.zph(fit1))
```

